Autosave Feature

Cthomp7 opened the pull request:
-Added autosave feature (created 'change' event listeners)
-Removed "update step" button and event listener
-Updated remove button styling to prevent easily clicking

nbriz commented on www/widgets/DemoExampleMaker.js:
```diff
@@ -86,6 +86,10 @@ class DemoExampleMaker extends Widget {
           display: flex;
           justify-content: space-between;
         }
+        #dem-remove-step:hover {
+          background-color: rgb(252, 97, 86);
+          border-color: rgb(252, 97, 86);
+        }
```
let's change the selector here to avoid using `id`, this is probably a specific enough id that it shouldn't cause issues w/future widgets, but just to be extra safe i've been using this overly cautious/specific-to-this-widget approach:

```css
let's remove this `style="float: right"` from here && add it the CSS class u created:

```css
.demo-example-maker  button[name="dem-remove-step"]:hover  {
  background-color: rgb(252, 97, 86);
  border-color: rgb(252, 97, 86);
}
```

nbriz commented on www/widgets/DemoExampleMaker.js:
```diff
@@ -99,8 +103,7 @@ class DemoExampleMaker extends Widget {
         <button name="dem-preview-step">preview</button>
-        <button name="dem-update-step">update this step</button>
-        <button name="dem-remove-step">remove this step</button>
+        <button id="dem-remove-step" name="dem-remove-step" style="float: right">remove this step</button>
```
given the above change, we can now remove the `id="dem-remove-step"` from this element

nbriz reviewed (changes requested):
this works great! tiny refactoring note below (just to avoid some potential future issues)

Cthomp7 commented:
@nbriz, I have removed the id and added the `float: right;` style up in the css, however since this should be displaying at all times and not just when being hovered I added a non-hover class above. 

Let me know if there are any other issues!
